pypechain/render/contract.py

```diff
@@ -488,3 +497,42 @@ def get_event_datas(abi: ABI) -> dict[str, EventData]:
             }
             event_datas[name] = event_data
     return event_datas
+
+
+def get_link_reference_data(link_references: list[LinkReferences]) -> LinkReferencesData:
+    """Gets the link reference data required for the contract template.
+
+    Parameters
+    ----------
+    link_references: list[LinkReferences]
+        A list of LinkReferences to be used in the contract template
+
+    Returns
+    -------
+    LinkReferencesData
+        A data structure containing the link reference data
+    """
+    contract_name_keys = []
+    contract_name_key_and_types = []
+    contract_types = []
+    contract_name_key_to_placeholder_lookup = []
+    for link in link_references:
+        contract_name = link["contract_name"]
+        # The contract_name_key is reformatting the contract name to be snake case
+        contract_name_key = camel_to_snake(contract_name)
```

Comment:
maybe a nit here, but I like when we preserve names of files and contracts 'as-is'.  it makes it more clear what we're trying to match when we provide addresses.